src/rules/strictErrorThrowRule.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isThrowStatement, isTypeFlagSet } from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+// tslint:disable:no-bitwise
+
+export class Rule extends Lint.Rules.TypedRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "strict-error-throw",
+        description: Lint.Utils.dedent`
+            Restricts that only instances of \`Error\` or its subclass be thrown.`,
```

Comment:
"...only instances or subclasses of `Error` be thrown"

Reply:
updated